solidity/ecdsa/contracts/WalletRegistry.sol

```diff
@@ -72,6 +80,22 @@ contract WalletRegistry is IRandomBeaconConsumer, IWalletRegistry, Ownable {
     ///         operator affected.
     uint256 public maliciousDkgResultNotificationRewardMultiplier;
 
+    /// @notice Calculated max gas cost for submitting a dkg result. This will
+    ///         be refunded as part of the dkg approval process. It is in the
+    ///         submitter's interest to not skip his priority turn on the approval,
+    ///         otherwise the refund of the dkg submission will be refunded to
+    ///         other member that will call the dkg approve function.
+    uint256 public dkgResultSubmissionGas = 305000;
```

Comment:
Is the problem you mentioned in the PR description about `submitDkgResult` or `approveDkgResult`?

The gas cost of `submitDkgResult` should be pretty much the same for all invocations and hardcoding it - assuming it is an updateable value - could make sense, given we avoid the additional cost of storing it.

On the other hand, the cost of `approveDkgResult` should vary and it could make more sense to measure it.

One option to avoid problems with the cost of `calldata` could be to let to pass an extra gas parameter to `refundable`. We could have two versions of the modifier: `modifier refundable(address receiver)` and `modifier refundable(address receiver, uint256 extraGas)`. That extra gas would also need to be governable.

Something to think about and measure with unit tests.

Reply:
see https://github.com/keep-network/keep-core/pull/2856#discussion_r830188379